src/core/credentials/transport/alts/alts_security_connector.cc

```diff
@@ -252,6 +260,11 @@ RefCountedPtr<grpc_auth_context> grpc_alts_auth_context_from_tsi_peer(
   size_t i = 0;
   for (i = 0; i < peer->property_count; i++) {
     const tsi_peer_property* tsi_prop = &peer->properties[i];
+    // This would be the case if an optional property was not provided, i.e.
+    // TSI_ALTS_NEGOTIATED_TRANSPORT_PROTOCOL
+    if (tsi_prop->name == nullptr) {
```

Comment:
We can double check with @matthewstevenson88 , but I'm thinking that this isn't the way we want to go - I could see this causing a subtle bug in the future because the unnamed prop is fairly non-specific.

I think rather properties should be required, and if the transport protocol negotiation doesn't happen it defaults to `http/2`. What do you think?

Reply:
Isn't this the behavior for our internal implementation? Whatever we think is the right thing to do, I think we should ensure that the two implementations match.

Also: it's possible that defaulting to `http/2` is the right thing to do, but if we do the defaulting then I think it should be wherever in gRPC is reading the transport protocol that does the defaulting, rather than the ALTS layer.

WDYT?